docker-jans-configurator/README.md

```diff
@@ -72,34 +72,34 @@ The following commands are supported by the container:
 
 The load command can be used either to generate or restore config and secret for the cluster.
 
-#### Docker
+For fresh installation, generate the initial configuration and secret by creating `/path/to/host/volume/generate.json` similar to example below:
+```json
+{
+    "hostname": "demoexample.jans.io",
+    "country_code": "US",
+    "state": "TX",
+    "city": "Austin",
+    "admin_pw": "S3cr3t+pass",
+    "ldap_pw": "S3cr3t+pass",
+    "email": "s@jans.io",
+    "org_name": "Gluu Inc."
+}
+```
 
-1. To generate the initial configuration and secret, create `/path/to/host/volume/generate.json` similar to example below:
-
-    ```json
-    {
-        "hostname": "demoexample.jans.io",
-        "country_code": "US",
-        "state": "TX",
-        "city": "Austin",
-        "admin_pw": "S3cr3t+pass",
-        "ldap_pw": "S3cr3t+pass",
-        "email": "s@jans.io",
-        "org_name": "Gluu Inc."
-    }
-    ```
+**NOTE**: `generate.json` has optional attributes as seen below.
 
-    **NOTE**: `generate.json` has optional attributes to generate oxAuth signing and encryption keys based on specific algorithms.
+- `auth_sig_keys`: space-separated key algorithm for signing (default to `RS256 RS384 RS512 ES256 ES384 ES512 PS256 PS384 PS512`)
+- `auth_enc_keys`: space-separated key algorithm for encryption (default to `RSA1_5 RSA-OAEP`)
+- `optional_scopes`: list of scopes that will be used (supported scopes are `ldap`, `scim`, `fido2`, `couchbase`, `redis`, `sql`, `casa`; default to empty list)
+- `ldap_pw`: user's password to access LDAP database (only used if `optional_scopes` list contains `ldap` scope)
+- `sql_pw`: user's password to access SQL database (only used if `optional_scopes` list contains `sql` scope)
+- `couchbase_pw`: user's password to access Couchbase database (only used if `optional_scopes` list contains `couchbase` scope)
+- `couchbase_superuser_pw`: superusers password to access Couchbase database (only used if `optional_scopes` list contains `couchbase` scope)
+- `salt`: user-defined salt (24 characters length); if omitted, salt will be generated automatically
```

Comment:
We should clearly define the regex  and type for the 24 char I.e byte, string , base64 decoder or plain text. We can use bash RANDOM as an example.

Reply:
good idea ... i will add more validation to ensure the value is correct

Reply:
alphanumeric validation has been added at https://github.com/JanssenProject/jans/pull/3624/commits/bb0d8f34d6d8d3b2338dcf2288e8aee29023d257